### ISSUE TITLE
cppcheck finds quite a few issues on the code

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#957]
**Reported by Carlos Lopez  clopez on 3 Dec 2011 17:52 UTC**

cppcheck finds quite a few issues:

```
[(error) Memory leak: ivs_table
[src/airodump-ng.c:6353](src/airdecloak-ng.c:1128]:): (error) Possible null pointer dereference: ap_cur - otherwise it is redundant to check if ap_cur is null at line 6355
[(error) Possible null pointer dereference: ap_cur - otherwise it is redundant to check if ap_cur is null at line 325
[src/airtun-ng.c:677](src/aircrack-ng.c:323]:): (error) Resource leak: f
[(error) Resource leak: f
[src/kstats.c:328](src/airbase-ng.c:1457]:): (error) Resource leak: f
[(error) Memory leak: ivbuf
[src/ivstools.c:168](src/kstats.c:322]:): (error) Resource leak: f_in
[(error) Resource leak: f_out
[src/ivstools.c:814](src/ivstools.c:162]:): (error) Resource leak: f_in
[(error) Resource leak: f_cap_out
[src/aireplay-ng.c:1162](src/aireplay-ng.c:1107]:): (error) Resource leak: f
[(error) Memory leak: key
[src/osdep/linux.c:956](src/aircrack-ng.c:4392]:): (error) Resource leak: fd2
[(error) Resource leak: fp
[src/osdep/linux.c:2074](src/airodump-ng.c:401]:): (error) Resource leak: apm
[(error) Memory leak: uiv_lvl2
[src/tkiptun-ng.c:1326](src/uniqueiv.c:117]:): (error) Resource leak: f_cap_out
[(error) Resource leak: f
[src/sha1-sse2.h:72](src/tkiptun-ng.c:1381]:): (error) Dangerous usage of 'essid' (strncpy doesn't always 0-terminate it)
[src/sha1-sse2.h:80]: (error) Dangerous usage of 'key1' (strncpy doesn't always 0-terminate it)
```

''http:''//''bugs.debian.org/cgi-bin/bugreport.cgi?bug=642934''

**Comment by aircrack-ng** [comment]
**Comment by misterx on 3 Dec 2011 17:52 UTC**

cppcheck is wrong for quite a few things.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 3 Dec 2011 17:52 UTC**

Fixed in aircrack-ng/aircrack-ng@a33074e72e5da5095f1d7f7d724c2da699b2af0c.
